test/AspNetCoreModule.TestSites.Standard/TestMiddleWareBeforeIISMiddleWare.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Security.Principal;
+using System.Threading.Tasks;
+
+
+namespace AspnetCoreModule.TestSites.Standard
+
+{
+    public class TestMiddleWareBeforeIISMiddleWare
+    {
+        private readonly RequestDelegate next;
+
+        public TestMiddleWareBeforeIISMiddleWare(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            // if the given request is shutdown message from ANCM and the value of GracefulShutdown environment variable is set,
+            // the shutdown message is handled by this middleware instead of IISMiddleware.
+
+            if (HttpMethods.IsPost(context.Request.Method) &&
+                context.Request.Path.ToString().EndsWith("/iisintegration") &&
```

Comment:
EndsWith? Why not just Path.Equals?

Reply:
That is because this test app is not placed on the root application. It is placed on a child application on IIS websites. In that case, the Request Path will look like "/<childappName/iisintegration.

Reply:
Oh, that's what the PathBase and UsePathBase was for.

Reply:
@Tratcher Does that mean we don't use the "UsePathBase"? In debugging, I noticed that the Request.PathBase was empty unlike Request.Path and I was wondering that happens because we did not use the flag in aspnetcore.

Reply:
@Tratcher Ah, I found UsePathBase on https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Abstractions/Extensions/UsePathBaseExtensions.cs.

Reply:
Yeah that one :) You can specify the pathbase for an app and run that.